% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newlength{\cslentryspacingunit} % times entry-spacing
\setlength{\cslentryspacingunit}{\parskip}
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
  \let\oldpar\par
  \def\par{\hangindent=\cslhangindent\oldpar}
  \fi
  % set entry spacing
  \setlength{\parskip}{#2\cslentryspacingunit}
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
\usepackage{xcolor, colortbl, rotating, graphicx, caption, subcaption}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={scLMM: fast and scalable single-cell differential expression analysis using linear mixed-effects models},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{scLMM: fast and scalable single-cell differential expression
analysis using linear mixed-effects models}
\author{}
\date{\vspace{-2.5em}October 16, 2024}

\begin{document}
\maketitle
\begin{abstract}
Single-cell RNA sequencing (scRNA-seq) allows detailed comparisons of
gene expression across individual cells and conditions. Single-cell
differential expression (DE) analysis faces challenges like sample
correlation, individual variation and scalability. We developed a fast
and scalable linear mixed-effects model (LMM) algorithm and scLMM R
package to address these issues. Our method leverages summary statistics
bypassing direct cell-level measurements for reduced computational
complexity and memory use. Simulation studies with scRNA-seq data
confirmed our algorithm's accuracy and efficiency, and scLMM effectively
controls false positives and maintains high statistical power in DE
analysis. Applications in real-world datasets, such as tuberculosis
immune atlas and kidney studies, demonstrate scLMM's utility in
accelerating single-cell studies across diverse biological contexts.
\end{abstract}

\hypertarget{background}{%
\section{Background}\label{background}}

Single-cell RNA sequencing (scRNA-seq) technology enables researchers to
compare the transcriptomes of individual cells and assess
transcriptional similarities and differences within a population. This
analysis is typically to identify differentially expressed genes (DEG)
and/or marker genes by comparing gene expressions in cell populations
under various biological conditions, such as health and disease, and/or
cell cluster annotation. The increased cost efficiency and throughput of
scRNA-seq now make it feasible to sequence a larger number of samples.
Data can be generated from samples comprising hundreds of subjects and
millions of cells, making differential expression (DE) analysis more
challenging. The key challenges include 1) correlation within subjects
and cell populations and high variability between subjects and across
cell populations, and 2) large scale of the data, consisting of hundreds
of thousands to millions of cells. Mixed-effect models have been used to
address the challenge of intra-subject correlation and inter-subject
variability by modeling the subject as a random effect (Hoffman and
Roussos (\protect\hyperlink{ref-Dream2020}{2020}), He et al.
(\protect\hyperlink{ref-nebula}{2021}), Zimmerman, Espeland, and
Langefeld (\protect\hyperlink{ref-Zimmerman2021}{2021}), Gagnon et al.
(\protect\hyperlink{ref-MSMCSim2022}{2022})). In multi-subject
single-cell studies, it is not feasible to model subjects as a fixed
effect when comparing individuals, where each subject is perfectly
correlated with a specific condition of interest. Due to computational
challenges, the performance of mixed-effects models for DE analyses has
mostly been examined through simulation analyses with only small numbers
of subjects and cells or pseudobulk methods.

To address the challenge of large-scale scRNA-seq in linear
mixed-effects model (LMM) based DE analysis, we developed a fast and
scalable LMM estimation algorithm and created an R package named scLMM,
the single-cell (sc) analysis by a linear mixed-effects model (LMM). Our
LMM estimation leverages summary statistics of the correlation
relationship between the response matrix and design matrices as inputs,
rather than using cell-level measurements directly. These summary
statistics can be computed quickly in advance. Once computed, the LMM
estimation computation is independent of the number of cells (sample
size). Comparing to the standard LMM estimation method lmer in lme4
package (\protect\hyperlink{ref-lme4}{Bates et al. 2015}), our LMM
algorithm is much fast and accurate, and requires significantly less
computer memory. Importantly, by the scLMM, we can comprehensively
examine the performance of LMM in the large scale single-cell data
analysis.

Our LMM estimation algorithm allows the parameters of variance
components to be negative such that the variance-covariance matrix of
observed response measurements is definable (positive definite). This
avoids zero boundary problem in the hypothesis testing of zero variance
components. Hence, the asymptotic property of maximum likelihood
estimate (MLE) holds under regularity conditions, and then we can use
z-statistic or t-statistic for hypothesis tests of fixed effects and
variance components. If the parameters of variance components are
negative or zero, this implies that the mixed model is miss-specified
and no random effects should be included.

We evaluated the performance of scLMM through simulation studies based
on metrics: 1) the accuracy and computational efficiency of our LMM
fitting algorithm, and 2) the type-I error rate and statistical power of
the LMM-based DE analysis method. For the simulation studies, we
developed a scRNA-seq simulator, named simuRNAseq, to generate a
multi-subject multi-cell-type scRNA-seq data based on a negative
binomial (NB) distribution. Our simulator is similar to muscat
(\protect\hyperlink{ref-muscat2020}{Crowell et al. 2020}) and GLMsim
(\protect\hyperlink{ref-GLMsim2024}{Wang et al. 2024}) but different in
the estimation of the NB parameters. Both muscat and GLMsim have
limitations in scRNA-seq simulations. The muscat estimates the
dispersion of NB distribution by edgeR package based on a subset of the
reference data, which cannot be applied in a large scale scRNA-seq data
and only reflects a partial information of the reference data. GLMsim
estimates coefficients and dispersion parameter of the NB model for each
gene by glm.nb in MASS package, which is time-consuming, and only
generates a data with a fixed size as the reference data. Our simuRNAseq
simulator computes the dispersion of NB distribution models by the
method-of-moments estimate (MME)
(\protect\hyperlink{ref-Clark1989}{Clark and Perry 1989}), which is fast
and flexible and uses a whole biological reference data.

We demonstrated the application of scLMM for case-control comparisons in
a tuberculosis immune atlas, cell-type-specific sex comparisons in
healthy kidney datasets, and the identification of cell-type marker
genes in both contexts. In summary, scLMM not only accelerates analysis
in single-cell studies but also enhances accuracy across diverse
biological contexts, paving the way for leveraging mixed models in
large-scale, multi-subject single-cell data analysis.

\hypertarget{results}{%
\section{Results}\label{results}}

\hypertarget{overview-of-sclmm}{%
\subsection{Overview of scLMM}\label{overview-of-sclmm}}

scLMM package comprises the main functions: lmm, lmmfit and lmmtest.
Both lmm and lmmfit functions are to fit LMM by restricted maximum
likelihood (REML). The two functions are identical in LMM fitting but
different in arguments. lmmfit directly uses response matrix and design
matrices as arguments, while lmm uses summary statistics of correlation
relationship between response matrix and design matrices as arguments.
The summary statistics can be fast computed in advance. lmm doesn't
depend on the sample size (the number of cells for scRNA-seq data) and
needs much less computer memory in the single-cell DE analysis. lmmtest
conducts t-tests for fixed effects and contrasts of the fixed effects.
The scLMM workflow for the DE analysis is shown in Figure
\ref{fig:scLMM}. See the LMM estimation and inference in Methods section
and Supplemental Materials.

\begin{figure}
\centering
\includegraphics{Figures/diagram_scLMM.pdf}
\caption{\textbf{scLMM workflow for single-cell differential expression
analysis}. A) Data: Gene expressions \(Y = log(1 + \text{counts})\).
Fixed effects include various variables such as log-library size, batch
effects, biological conditions of interest, and interactions between
conditions and cell-types, while random effects account for variations
between individual subjects. B) Model: Define the linear mixed-effects
model (LMM) for each gene by design matrices \(X\) and \(Z\), which are
constructed based on prior knowledge about the covariates and the
biological question. C) Model fitting: Fit the LMM using either lmm or
lmmfit. Summary statistics are computed as follows: \(XX = X^TX\),
\(XY = X^TY^T\), \(ZX = Z^TX\), \(ZY = Z^TY^T\), and \(ZZ = Z^TZ\).
\(Ynorm = diag(YY^T)\) and \(n\) is the number of cells. The output
includes estimates of coefficients, t-values and p-values for the fixed
effects, the covariance matrix of the estimated coefficients, and
variance components. Use lmmtest to perform statistical tests on the
fixed effects and contrasts based on the LMM
estimates.\label{fig:scLMM}}
\end{figure}

\hypertarget{simulation-studies}{%
\subsection{Simulation studies}\label{simulation-studies}}

We validated the accuracy and computational efficiency of lmm and lmmfit
by comparing to the standard method lmer in lme4 package
(\protect\hyperlink{ref-lme4}{Bates et al. 2015}) based on simulated
scRNA-seq data. We also evaluate the performance of lmm/lmmfit in
single-cell DE analysis through simulations by criteria: (1) control of
Type-I-error (False positive rate or FPR) and (2) statistical power
(true positive rate or TPR) with comparing to the NEBULA
(\protect\hyperlink{ref-nebula}{He et al. 2021}). We developed a
scRNA-seq simulator, named simuRNAseq, to simulate a multi-subject
multi-cell-type scRNA-seq data, by using a reference data based on a
negative binomial (NB) distribution, see the section of simulation
methods in Supplemental Materials. The simuRNAseq simulator computes the
dispersion of NB distribution models by the method-of-moments estimate
(MME) (\protect\hyperlink{ref-Clark1989}{Clark and Perry 1989}).
Compared to the maximum likelihood based estimates, such as glm.nb, the
MME is computationally more simple and performs well.

\textbf{Simulating scRNA-seq data}: We used PBMC 10X droplet-based
scRNA-seq data from lupus patients
(\protect\hyperlink{ref-PBMC2018}{Kang et al. 2018}), as a reference to
simulate scRNA-seq datasets with 6,000 genes and 6 sample sizes from
20,000 to 120,000 cells. The genes were randomly selected from the
reference data, and the cells were simulated from 25 subjects and 12
cell-types treated with two conditions. The treatments, cell-types and
subjects are assigned randomly with equal probability. There are 480 DE
genes specific to a cell-type.

\textbf{Accuracy and computational efficiency}: We first validated the
accuracy and computational efficiency of lmm and lmmfit using simulated
datasets by comparing to the lmer in lme4 package
(\protect\hyperlink{ref-lme4}{Bates et al. 2015}). We fit the LMM to the
log-transformed counts, \(log(1 + counts)\), respectively, by lmm,
lmmfit, and lmer with default setting. Note that the lmer in lme4
package doesn't provide p-values for hypothesis tests of coefficients.
We refit the LMM by the lmer in lmerTest package
(\protect\hyperlink{ref-lmerTest}{Kuznetsova, Brockhoff, and Christensen
2017}) to obtain the p-values in the lmer fitting. lmerTest package
overloads lmer from lme4 package and extends output of the summary
method from the lme4 package by adding degrees of freedom using the
Satterthwaite's or Kenward-Roger's approximations for the t test and
corresponding p values.

The differences of variance components, coefficients, and p-values
between lmm and lmer are shown in Figure \ref{fig:lmer} (a). The model
parameters (coefficients and variance components) in lmm and lmer
fittings are identical up to the sixth digit. Note that the differences
between lmm and lmer fittings can be further reduced by changing the
control parameter in lmer to increase the accuracy of lmer fitting, but
the lmer computation time will also increase. The computation time for
running lmm, lmmfit and the lmer with default setting in a MacBook with
16GB memory is shown in Figure \ref{fig:lmer} (b). It is seen that lmm
and lmmfit are much faster than lmer. lmmfit was about 50-fold to
140-fold faster than lmer as sample size increases from 20,000 to
120,000. lmm took much least time about 0.3 minutes and the time almost
didn't change as sample sizes increases. The exact running times are
listed in Supplemental Table.

\textbf{Performance in single-cell DE analysis}: The QQ-plots of non-DE
genes p-values and receiver operating characteristic (ROC) curves at
sample size \(n=120000\) for lmm and nebula are shown in Figure
\ref{fig:lmer} (c) and (d). The nebula was run with arguments:
method=`LN' and model=`NBLMM', the negative binomial lognormal mixed
model that is used for simulating the data. The QQ-plots show that lmm
has a good control of Type-I-error while nebula slightly deflates the
p-values. The ROC curves show that lmm has a close power with nebula.
The QQ-plots and ROC curves across various sample sizes are shown in
Supplemental Figures. The scatterplots in Supplemental Figures also show
that lmm and nebula t-values and p-values are coincident.

\begin{figure}[htb]
     \centering
     \begin{subfigure}[b]{0.45\textwidth}
         \centering
         \includegraphics[width=\textwidth]{Figures/simuNBMM_lmmfitdiff_p.png}
         \caption{Differences of LMM estimates}
     \end{subfigure}
     \begin{subfigure}[b]{0.45\textwidth}
         \centering
         \includegraphics[width=\textwidth]{Figures/simuNBMM_runtime3.pdf}
         \caption{Computation time}
     \end{subfigure}
     \begin{subfigure}[b]{0.45\textwidth}
         \centering
         \includegraphics[width=1.01\textwidth, height = 1.01\textwidth]{Figures/simuNBMM_qqplot_null120000.pdf}
         \caption{QQ-plot}
     \end{subfigure}
     \begin{subfigure}[b]{0.45\textwidth}
         \centering
         \includegraphics[width=\textwidth]{Figures/simuNBMM_AUC120000.pdf}
         \caption{ROC curve}
     \end{subfigure}
     \centering
     \caption{\textbf{Computational and statistical performance of lmm in differntial expression analysis of simulated scRNA-seq data}. (a) Boxplots of differences of variance components, coefficients, and p-values between lmm and lmer fitting across various sample sizes. (b) Computation time (in minutes) across different sample sizes for lmm, lmmfit, lmer and nebula. (c) QQ-plots of non-DE genes p-values for lmm and nebula. (d) ROC curves for lmm and nebula.}
     \label{fig:lmer}
\end{figure}

\hypertarget{de-analysis-of-biological-scrna-seq-data}{%
\subsection{DE analysis of biological scRNA-seq
data}\label{de-analysis-of-biological-scrna-seq-data}}

\textbf{Kidney scRNA-seq data}: We first examined the sex variations
within healthy kidney cell types using the kidney scRNA-seq data
{[}citation{]}. The kidney data from 19 subject samples contains 22,484
genes and 27,677 cells consisting of 18 cell-types. After filtering in
quality control process, the data contains 14,175 genes and 27,550
cells. We performed a differential expression analysis using scLMM to
identify the DE genes between male and female within a cell-type with
taking into account the subjects as a random effect. Among the various
cell populations (types), we highlight the cortical thick ascending limb
(cTAL) cells, which are the second most abundant cell type and exhibit
131 DE genes with FDR \textless{} 0.05 and logFC \textless{} -0.25 or
\textgreater{} 0.25. The top DE genes in male and female cTAL cells are
shown in Figure \ref{fig:kidneydeg} A and B. Pathway analysis of the top
DE genes revealed enrichment of angiogenesis and vasculature development
pathways in male cTAL cells, and cytoskeleton and filament assembly
components in female cTAL cells, Figure \ref{fig:kidneydeg} C and D.

For comparison, we also performed the DE analysis using the lmer with
default setting. The differences between lmm and lmer fittings and
computation time for lmm, lmmfit and lmer are shown in the Biological
Data Analysis section of supplemental materials. The estimated model
parameters (coefficients and variance components) in lmm and lmer
fittings are identical up to at least the sixth digit. lmm is about
100-fold faster than lmer for fitting the kidney data.

\begin{figure}
\centering
\includegraphics{Figures/kidney-DEG.pdf}
\caption{\textbf{scLMM identifies sex-specific variations in a healthy
human kidney map}. A) The top male-specific DE genes within the cTAL
population. B) The top female-specific DE genes within the cTAL
population. C) Pathway enrichment results based on the male-specific DE
genes in the cTAL population. D) Pathway enrichment results based on the
female-specific DE genes in the cTAL population.\label{fig:kidneydeg}}
\end{figure}

\textbf{Tuberculosis (TB) scRNA-seq data}: We then applied scLMM to
single-cell transcriptomics data from 500K memory T cells from 259
donors in a tuberculosis (TB) progression cohort
(\protect\hyperlink{ref-Nathan2021}{Nathan et al. 2021}). After quality
control process, the large TB dataset contains 11,596 genes and 499,713
cells covering 29 cell states from 46 batches and 259 individual donors.
The majority of the samples have been sequenced in a single batch. A few
of them have been split into two batches. In the DE analysis, we modeled
the donors as a random effect and ignored the batch effect because the
majority of donors were sequenced in a single batch. We applied scLMM to
identify genes associated with TB status within each cell state
(cluster). scLMM identified a varying number of DE genes associated with
TB progression across different cell states (FDR \textless{} 0.05 and
logFC \textgreater{} 0). The top two cell types with the highest number
of DE genes were the activated CD4+ and activated CD8+ populations (The
numbers of DE genes: 1266 and 268, respectively; Figure
\ref{fig:TBdeg}A). We further evaluated the TB-associated signatures
within these two cell states, listing the top TB-enriched genes for
activated CD4+ and CD8+ cells in Figure \ref{fig:TBdeg}B. Pathway
analysis of these DE genes revealed enrichment of cell-cycle pathways in
the CD4+ population and immune response, TCR-mediated T cell activation,
and chemokine secretion pathways in the CD8+ cell state, Figure
\ref{fig:TBdeg}C.

The estimated model parameters (coefficients and variance components) in
lmm and lmer fittings are identical up to at least the sixth digit.
Notably, scLMM took about 1.4 hours of runtime on the 500K T cell
dataset, whereas lmer spent 55.6 hours (2 days and 7.6 hours), see the
Biological Data Analysis section in supplemental materials. This
indicates that for large datasets, scLMM is the more viable option.

\begin{figure}
\centering
\includegraphics{Figures/TB-DEG.pdf}
\caption{\textbf{scLMM identifies TB-enriched signatures within T cell
populations while accounting for confounding variables}. A) Bar plots
indicate the number of DE genes for each cell type. B) The top
TB-associated DE genes within the CD4+ activated and CD8+ activated T
cell populations are identified. C) Pathway enrichment results are
presented for the TB-enriched DE genes within the activated CD8+ and
CD4+ T cell populations.\label{fig:TBdeg}}
\end{figure}

\hypertarget{conclusion-and-discussion}{%
\section{Conclusion and discussion}\label{conclusion-and-discussion}}

We have developed a fast and scalable LMM estimation algorithm and
created scLMM for large scale single-cell DE analysis. The simulation
studies demonstrated that our LMM algorithm is both accurate and
computationally efficient compared to the standard lmer method in the
lme4 package. Our algorithm proved to be hundreds of times faster than
lmer for the simulated scRNA-seq data with hundreds of thousands of
cells. Our simulation studies verified that the LMM-based DE analysis
method can effectively control false positive rate while maintaining
high statistical power. Additionally, we applied scLMM to the
large-scale tuberculosis dataset, demonstrating its ability to identify
marker genes and facilitate group-based comparisons.

Our LMM estimation algorithm leverages summary statistics rather than
using cell-level measurements directly. These summary statistics can be
computed quickly in advance. Once computed, the LMM estimation achieves
a computational complexity of \(O\{m(p^3 +q^3)\}\), which is independent
of the sample size \(n\) (the number of cells). In the DE analysis, the
numbers of fixed and random effects, \(p\) and \(q\), are relatively
small, and consequently the LMM algorithm is fast and requires
significantly less computer memory. In the application with a large
\(q\), the LMM algorithm will become slow. In this case, we should first
reduce the dimension of random effects by cluster analysis or principal
component analysis (PCA), see Supplemental Materials for details. Due to
the conditions of LMM estimability and overfitting, the number of fixed
effects, \(p\), is not allowed to be large. This is also why we should
treat subjects and/or batches as random effects instead of fixed effects
when these numbers are large.

Using scLMM, we can comprehensively examine the performance of LMM in
the large-scale scRNA-seq DE analysis. For future work, a benchmark
comparing LMM with cell-level counts, pseudobulk method with
subject-level pseudobulk counts (Zimmerman, Espeland, and Langefeld
(\protect\hyperlink{ref-Zimmerman2021}{2021}), Gagnon et al.
(\protect\hyperlink{ref-MSMCSim2022}{2022})), and generalized linear
mixed model (GLMM) based methods (e.g., NEBULA, He et al.
(\protect\hyperlink{ref-nebula}{2021})) across various experimental
settings could provide valuable insights.

The mixed models have become powerful tools in data analysis due to the
capability of modeling of the inter-subject variation and intra-subject
correlation. scLMM is versatile and indicates potential application with
other data modalities such as spatial data and multiomics. Its efficient
mixed model framework, designed to handle complex hierarchical
structures and correlations in single-cell RNA sequencing data, suggests
it could effectively extend to these domains. Further exploration of
scLMM's capabilities across diverse biological datasets holds promise
for uncovering novel insights and enabling integrated analyses in
broader research contexts.

\hypertarget{methods}{%
\section{Methods}\label{methods}}

\hypertarget{lmm-estimation-and-inference}{%
\subsection{LMM estimation and
inference}\label{lmm-estimation-and-inference}}

Consider the linear mixed-effects model (LMM) as expressed below
(\protect\hyperlink{ref-Searle2006}{Searle, Casella, and McCulloch
2006}) \begin{equation} \label{lmm}
y = X\beta + Zb + \epsilon,
\end{equation} where \(y\) is an \(n\times 1\) vector of observed
responses (expressions for a gene) , \(X\) is an \(n\times p\) design
matrix for fixed effects \(\beta\), \(Z\) is an \(n\times q\) design
matrix for random effects \(b\), and \(\epsilon\) is an \(n\times 1\)
vector of residual errors. The term of random effects may be a
combination of various effects \[
Zb = Z_1 b_1 + \cdots + Z_K b_K,
\] where \(Z=[Z_1,\ldots,Z_K]\), \(b=[b^T_1,\ldots,b^T_K]^T\), \(K\) is
the number of the various random effects occurring in the data, and
\(Z_i\) is an \(n\times q_i\) matrix. The superscript \(T\) denotes a
transpose of vector or matrix. The basic assumptions are as follows: (1)
The design matrix \(X\) is of full rank, satisfying conditions of
estimability for the parameters; (2) The random vectors \(b_i\) and
\(\epsilon\) are independent and follow a normal distribution,
\(b_i \sim N(\mathbf{0}, \sigma^2_i I_{q_i})\) and
\(\epsilon \sim N(\mathbf{0}, \sigma^2I_n)\). Here \(\sigma^2_i\) and
\(\sigma^2\) are unknown parameters, called variance components,
\(\mathbf{0}\) is a vector or matrix of zero elements, and \(I_n\) is an
\(n\times n\) identity matrix. The random effects reflect variations
between groups (subjects) and correlations within groups (subjects).

Hartley and Rao (\protect\hyperlink{ref-HartleyRao1967}{1967}) developed
maximum likelihood (ML) method for estimation of the unknown fixed
effects and variance components in the linear mixed-effects models. The
ML method estimates all parameters of fixed effects and variance
components together. Patterson and Thompson
(\protect\hyperlink{ref-PattersonThompson1971}{1971}) proposed a
modified maximum likelihood procedure which partitions the data into two
mutually uncorrelated parts, one being free of the fixed effects used
for estimating variance components, called restricted maximum likelihood
(REML) estimators. The REML estimator is unbiased. The MLE of variance
components is biased in general. Both methods are asymptotically
identical for estimating variance components and identical for
estimating fixed effects.

Estimating the variance components by either MLE or REML is a difficult
numerical problem. Various iterative methods based on the log
likelihood, called gradient methods, have been proposed
(\protect\hyperlink{ref-Searle2006}{Searle, Casella, and McCulloch
2006}). The gradient methods are represented by the iteration equation
\begin{equation}\label{gradient}
\theta^{(m+1)} = \theta^{(m)} + \Gamma(\theta^{(m)})\frac{\partial l(\theta^{(m)})}{\partial\theta},
\end{equation} where \(\partial l(\theta)/\partial\theta\) is the
gradient of the log likelihood function, and \(\Gamma(\theta)\) is a
modifier matrix of the gradient direction, see the details in the
Supplementary materials.

\textbf{Fast and scalable algorithm}: We developed a summary statistics
based algorithm for implementing the gradient method \eqref{gradient},
which uses summary-level matrices, \(X^TX\), \(X^TZ\) and \(Z^TZ\), to
estimate LMM parameters instead of individual-level matrices, \(X\) and
\(Z\), see the Supplementary materials. The algorithm has a complexity
of \(O(n(p^2 + q^2)+p^3+q^3)\). In single-cell DE analysis, the number
of cells (sample size) \(n\) is large while the numbers of fixed and
random effects \(p\) and \(q\) are relatively small, e.g., less than
hundreds. Thus the summary statistics based algorithm has a complexity
of \(O(n(p^2+q^2))\), linearly scalable with the sample size \(n\). The
summary statistics can be computed in advance. Once computed, the
algorithm has a complexity of \(O(p^3+q^3)\) that doesn't depend on the
sample size \(n\). So the algorithm is fast and requires less computer
memory in the single-cell DE analysis. The algorithm can further speed
up by reducing the dimension of random effects. For a large number of
random effects, using cluster analysis or principal component analysis
(PCA), we may combine the correlated random effects to reduce the
dimension, see the Supplementary materials.

\textbf{Hypothesis testing}: The hypothesis testing for fixed effects
and variance components can be respectively defined as \[
H_{0, i}: \beta_i = 0 ~~\text{versus}~~H_{1,i}: \beta_i\ne 0,
\] \[
H_{0, k}: \sigma^2_k = 0 ~~\text{versus}~~H_{1,k}: \sigma^2_k > 0.
\] The variance components under null hypothesis, \(\sigma^2_k=0\), are
on the boundary of the parameter space, in which case the MLE asymptotic
property is inappropriate. With reparameterizing the variance
components, \(\theta_k = \sigma^2\gamma_k\), the covariance matrix,
\(V_{\theta} = \sigma^2(I + \gamma_1Z_1Z_1^T + \ldots + \gamma_KZ_KZ_K^T)\),
is positive-definite and well-defined when
\(\gamma_k > - 1/\lambda_{max}\), where \(\lambda_{max} > 0\), is the
largest singular value of \(ZZ^T\). Now the parameters of variance
components, \(\theta_k\), can be negative. Then the hypotheses for the
variance components are extended as \[
H_{0, k}: \theta_k \le 0 ~~\text{versus}~~H_{1,k}: \theta_k > 0,
\] in which the zero components, \(\theta_k=0\), are no longer on the
boundary of the parameter space and the MLE asymptotic properties hold.
Then we can use z-statistic or t-statistic for hypothesis testing of
fixed effects and variance components. The t-statistics for fixed
effects are given by \begin{equation}\label{tcoef}
T_i = \frac{\hat\beta_i}{\sqrt{var(\hat\beta_i)}} = \frac{\hat\beta_i}{\sqrt{var(\hat\beta)_{ii}}} ~\sim ~t(n - p),
\end{equation} where \(var(\hat\beta) = (X^TV_{\theta}^{-1}X)^{-1}\), is
the covariance matrix of \(\hat\beta\). The t-statistic for a contrast,
a linear combination of the estimated fixed effects, \(c^T\hat\beta\),
is \begin{equation}\label{tcontrast}
T_c = \frac{c^T\hat\beta}{\sqrt{c^Tvar(\hat\beta) c}} \sim t(n-p).
\end{equation} The z-statistics for the parameters of variance
components are given by \begin{equation}\label{zvarcomp}
Z_k = \frac{\hat\theta_k}{\sqrt{[I(\hat\theta)^{-1}]_{kk}}} \sim N(0, 1),
\end{equation} where \(I(\theta)\) is the Fisher information matrix. If
\(Z_k > 0\), then \(\sigma_k^2 = \theta_k\) is definable and the mixed
model is well-specified. Otherwise, if \(Z_k \le 0\), the mixed model is
miss-specified, that is, no random effects are needed.

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-lme4}{}}%
Bates, Douglas, Martin Mächler, Ben Bolker, and Steve Walker. 2015.
{``Fitting Linear Mixed-Effects Models Using {lme4}.''} \emph{Journal of
Statistical Software} 67 (1): 1--48.
\url{https://doi.org/10.18637/jss.v067.i01}.

\leavevmode\vadjust pre{\hypertarget{ref-Clark1989}{}}%
Clark, Suzanne J., and Joe N. Perry. 1989. {``Estimation of the Negative
Binomial Parameter k by Maximum Quasi-Likelihood.''} \emph{Biometrics}
45 (1). \url{https://doi.org/10.2307/2532055}.

\leavevmode\vadjust pre{\hypertarget{ref-muscat2020}{}}%
Crowell, Helena L., Charlotte Soneson, Pierre-Luc Germain, Daniela
Calini, Ludovic Collin, Catarina Raposo, Dheeraj Malhotra, and Mark D.
Robinson. 2020. {``Muscat Detects Subpopulation-Specific State
Transitions from Multi-Sample Multi-Condition Single-Cell
Transcriptomics Data.''} \emph{Nature Communications} 11.
\url{https://doi.org/10.1038/s41467-020-19894-4}.

\leavevmode\vadjust pre{\hypertarget{ref-MSMCSim2022}{}}%
Gagnon, Jake, Lira Pi, Matthew Ryals, Qingwen Wan, Wenxing Hu, Zhengyu
Ouyang, Baohong Zhang, and Kejie Li. 2022. {``Recommendations of
scRNA-Seq Differential Gene Expression Analysis Based on Comprehensive
Benchmarking.''} \emph{Life} 12 (6).
\url{https://doi.org/10.3390/life12060850}.

\leavevmode\vadjust pre{\hypertarget{ref-HartleyRao1967}{}}%
Hartley, H. O., and J. N. K. Rao. 1967. {``Maximum-Likelihood Estimation
for the Mixed Analysis of Variance Model.''} \emph{Biometrika} 54 (1/2):
93--108. \url{http://www.jstor.org/stable/2333854}.

\leavevmode\vadjust pre{\hypertarget{ref-nebula}{}}%
He, Liang, Jose Davila-Velderrain, Tomokazu S. Sumida, David A. Hafler,
Manolis Kellis, and Alexander M. Kulminski. 2021. {``NEBULA Is a Fast
Negative Binomial Mixed Model for Differential or Co-Expression Analysis
of Large-Scale Multi-Subject Single-Cell Data.''} \emph{Communications
Biology} 4 (629). \url{https://doi.org/10.1038/s42003-021-02146-6}.

\leavevmode\vadjust pre{\hypertarget{ref-Dream2020}{}}%
Hoffman, Gabriel E, and Panos Roussos. 2020. {``{Dream: powerful
differential expression analysis for repeated measures designs}.''}
\emph{Bioinformatics} 37 (2): 192--201.
\url{https://doi.org/10.1093/bioinformatics/btaa687}.

\leavevmode\vadjust pre{\hypertarget{ref-PBMC2018}{}}%
Kang, Hyun Min, Meena Subramaniam, Sasha Targ, Michelle Nguyen, Lenka
Maliskova, Elizabeth McCarthy, Eunice Wan, et al. 2018. {``Multiplexed
Droplet Single-Cell RNA-Sequencing Using Natural Genetic Variation.''}
\emph{Nature Biotechnology} 36 (89--94).
\url{https://doi.org/10.1038/nbt.4042}.

\leavevmode\vadjust pre{\hypertarget{ref-lmerTest}{}}%
Kuznetsova, Alexandra, Per B. Brockhoff, and Rune H. B. Christensen.
2017. {``{lmerTest} Package: Tests in Linear Mixed Effects Models.''}
\emph{Journal of Statistical Software} 82 (13): 1--26.
\url{https://doi.org/10.18637/jss.v082.i13}.

\leavevmode\vadjust pre{\hypertarget{ref-Nathan2021}{}}%
Nathan, Aparna, Jessica I. Beynor, Yuriy Baglaenko, Sara Suliman,
Kazuyoshi Ishigaki, Samira Asgari, Chuan-Chin Huang, et al. 2021.
{``Multimodally Profiling Memory t Cells from a Tuberculosis Cohort
Identifies Cell State Associations with Demographics, Environment and
Disease.''} \emph{Nature Immunology} 22 (6): 781--93.
\url{https://doi.org/10.1038/s41590-021-00933-1}.

\leavevmode\vadjust pre{\hypertarget{ref-PattersonThompson1971}{}}%
Patterson, H. D., and R. Thompson. 1971. {``Recovery of Inter-Block
Information When Block Sizes Are Unequal.''} \emph{Biometrika} 58 (3):
545--54. \url{https://doi.org/10.2307/2334389}.

\leavevmode\vadjust pre{\hypertarget{ref-Searle2006}{}}%
Searle, Shayle R., George Casella, and Charles E. McCulloch. 2006.
\emph{Variance Components}. New Jersey: John Wiley \& Sons, Inc.

\leavevmode\vadjust pre{\hypertarget{ref-GLMsim2024}{}}%
Wang, Jianan, Lizhong Chen, Rachel Thijssen, Belinda Phipson, and
Terence P. Speed. 2024. {``GLMsim: A GLM-Based Single Cell RNA-Seq
Simulator Incorporating Batch and Biological Effects.''} \emph{bioRxiv}.
\url{https://doi.org/10.1101/2024.03.20.586030}.

\leavevmode\vadjust pre{\hypertarget{ref-Zimmerman2021}{}}%
Zimmerman, Kip D., Mark A. Espeland, and Carl D. Langefeld. 2021. {``A
Practical Solution to Pseudoreplication Bias in Single-Cell Studies.''}
\emph{Nature Communications} 12 (1): 738.
\url{https://doi.org/10.1038/s41467-021-21038-1}.

\end{CSLReferences}

\end{document}
